### PR TITLE
Add pre-commit and follow REUSE recommendations for cookiecutter.

### PR DESCRIPTION
**Pull request by Flameeyes**
This includes both license tagging the cookiecutter support and the template, and adding pre-commit configuration for this repository and the template itself.

### CODE REVIEW

File: CODE_OF_CONDUCT.md [hunk 1]
```diff
@@ -1,3 +1,9 @@
+<!--
+SPDX-FileCopyrightText: 2017 Scott Shawcroft, written for Adafruit Industries
+
+SPDX-License-Identifier: MIT
```
Comment (tannewt) [inline CODE_OF_CONDUCT.md:4]: This is incorrect. We based it on the Contributor Covenant which is CC-BY 4.0: https://www.contributor-covenant.org/ (at the bottom)

Also sourced from here: https://github.com/adafruit/Adafruit_Community_Code_of_Conduct

**Comment by tannewt** [comment]
@kattni and @sommersoft you probably want to refine the copyrights to include yourselves too.

File: CODE_OF_CONDUCT.md (hunk 1 above)
Comment (Flameeyes) [reply CODE_OF_CONDUCT.md:4]: Yeah this sounds like it needs to be updated. I'll update it in that repo, and also in TestRepo where this was overlooked.

Comment (Flameeyes) [reply CODE_OF_CONDUCT.md]: Sent https://github.com/adafruit/Adafruit_Community_Code_of_Conduct/pull/5 for tagging the code of conduct file.

Sent https://github.com/adafruit/Adafruit_CircuitPython_TestRepo/pull/34 for the test repo.

And updated on this one.

**Pull request merged**